R/conditions.R

```diff
@@ -0,0 +1,120 @@
+# These two functions only exist to create artifacts on the stack trace
+# (sys.call()) that instruct the stack trace pretty printer what parts of
+# the stack trace are interesting or not. The initial state is 1 and we
+# walk from the outermost call inwards. Each ..stacktraceoff.. decrements
+# the state by one, and each ..stacktraceon.. increments the state by one.
+# Any stack trace frame whose value is less than 1 is hidden, and finally,
+# the ..stacktraceon.. and ..stacktraceoff.. calls themselves are hidden
+# too.
+#
+# Example:
+# ..stacktraceoff..({
+#   tryCatch(
+#     withVisible(..stacktraceon..(userFunc()))
+#   )
+# })
+..stacktraceon.. <- identity
```

Comment:
I think these would be better as `..stacktraceon.. <- function(x) x`, so that R will keep source refs to them. With `identity`, there's no source ref.